src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -0,0 +1,298 @@
+                    case Nodes.ThisExpression:
+                        break;
+                    case Nodes.ThrowStatement:
+                        break;
+                    case Nodes.TryStatement:
+                        break;
+                    case Nodes.UnaryExpression:
+                        break;
+                    case Nodes.UpdateExpression:
+                        break;
+                    case Nodes.VariableDeclaration:
+                        break;
+                    case Nodes.VariableDeclarator:
+                        break;
+                    case Nodes.WhileStatement:
+                        break;
+                    case Nodes.WithStatement:
+                        break;
+                    case Nodes.ArrayPattern:
+                        break;
+                    case Nodes.AssignmentPattern:
+                        break;
+                    case Nodes.SpreadElement:
+                        break;
+                    case Nodes.ObjectPattern:
+                        break;
+                    case Nodes.ArrowParameterPlaceHolder:
+                        break;
+                    case Nodes.MetaProperty:
+                        break;
+                    case Nodes.Super:
+                        break;
+                    case Nodes.TaggedTemplateExpression:
+                        break;
+                    case Nodes.YieldExpression:
+                        break;
+                    case Nodes.ArrowFunctionExpression:
+                        break;
+                    case Nodes.ClassBody:
+                        break;
+                    case Nodes.ClassDeclaration:
+                        break;
+                    case Nodes.ForOfStatement:
+                        break;
+                    case Nodes.MethodDefinition:
+                        break;
+                    case Nodes.ImportSpecifier:
+                        break;
+                    case Nodes.ImportDefaultSpecifier:
+                        break;
+                    case Nodes.ImportNamespaceSpecifier:
+                        break;
+                    case Nodes.ImportDeclaration:
+                        break;
+                    case Nodes.ExportSpecifier:
+                        break;
+                    case Nodes.ExportNamedDeclaration:
+                        break;
+                    case Nodes.ExportAllDeclaration:
+                        break;
+                    case Nodes.ExportDefaultDeclaration:
+                        break;
+                    case Nodes.ClassExpression:
+                        break;
+                }
+            }
+        }
+
+        private IEnumerable<string> ExtractFields(JsValue map)
+        {
+            var bluh = map.ToString();
+            
+            if (map == null)
+                yield break;
+        }
+
+        private IEnumerable JintMapFuncWrapper(List<JsValue> jsValues, IEnumerable<dynamic> items)
```

Comment:
The name `jsValues` and the passing of `List<JsValue>` is really confusing. Better to pass `List<FunctionInstance>` here.